platform/darwin/src/MGLFeature.mm

```diff
@@ -5,14 +5,105 @@
 #import "MGLPolygon.h"
 #import "MGLValueEvaluator.h"
 
+#import "MGLShape_Private.h"
 #import "MGLMultiPoint_Private.h"
+#import "MGLPolyline+MGLAdditions.h"
+#import "MGLPolygon+MGLAdditions.h"
 
 #import <mbgl/util/geometry.hpp>
+#import <mapbox/geometry/feature.hpp>
 
-@protocol MGLFeaturePrivate <MGLFeature>
+@interface MGLFeatureUtility : NSObject
 
-@property (nonatomic, copy, nullable, readwrite) id identifier;
-@property (nonatomic, copy, readwrite) NS_DICTIONARY_OF(NSString *, id) *attributes;
++ (mbgl::Value)convertedValueWithValue:(id)value;
+
+@end
+
+@implementation MGLFeatureUtility
+
+// TODO: This is copied from NSExpression+MGLAdditions. Where should a shared implementation go?
```

Comment:
How about stuffing the value in an NSExpression and calling the existing category method?